lib/src/observable_list.dart

```diff
@@ -268,6 +268,26 @@ class ObservableList<E> extends ListBase<E> with ChangeNotifier {
     return false;
   }
 
+  ObservableList map(f(E element)) {
+    final derivedList = new ObservableList.from(this.map(f));
+
+    listChanges.listen((List<ListChangeRecord> crs) =>
```

Comment:
Another issue here that I was just thinking about is this creates a memory leak. The listener here is always going to have a reference to your derivedList. Since there is no way to cancel the listener (currently) then every call to map will create an event listener and a list which are doing potentially a fair bit of work and can never be cleaned up. See my proposed solution to this in the main comments (shortly).